### PR TITLE
otelcolconvert: support converting jaegerremotesampling extension

### LINKED ISSUE #6421: Support converting `jaegerremotesampling` extension

### PR DESCRIPTION
**Pull request by tpaschalis**
Closes #6421

cc @rfratto I saw you'd self-assigned the previous issue so I hope I'm not duplicating any work, but I wanted to get the rest of the static mode converters down.

### CODE REVIEW

File: internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river [hunk 1]
```diff
@@ -0,0 +1,42 @@
+otelcol.extension.jaeger_remote_sampling "default" {
+	grpc {
+		endpoint = ":14250"
+	}
+
+	http {
+		endpoint = ":5778"
+	}
```
Comment (rfratto) [inline internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river:8]: Why don't these match with the defaults anymore? Do we need a new issue to sync them?

**Review by rfratto — APPROVED** [review]
LGTM after comment about default drift

File: internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river (hunk 1 above)
Comment (tpaschalis) [reply internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river:8]: Not sure, let me look (I copied these over from the jaeger receiver testcase).

Comment (tpaschalis) [reply internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river:8]: Yeah, sorry for the confusion. I didn't realize the reason why we aliased the types in the package was that we could set up component-specific defaults. Fixing this now.

Comment (tpaschalis) [reply internal/converter/internal/otelcolconvert/testdata/jaegerremotesampling.river:8]: More specifically, we set up our defaults as 

https://github.com/grafana/agent/blob/2603bda1660254f57f22e42ac3b47ab2ef58eb6e/internal/component/otelcol/extension/jaeger_remote_sampling/jaeger_remote_sampling.go#L41-L50

whereas our fork of the jaegerremotesampling processor sets them up as
https://github.com/grafana/agent/blob/2603bda1660254f57f22e42ac3b47ab2ef58eb6e/internal/component/otelcol/extension/jaeger_remote_sampling/internal/jaegerremotesampling/factory.go#L44-L52

Upstream uses this function to decide whether to add localhost or 0.0.0.0 up front, so in a sense our defaults _did_ drift, and I'll add a commit to fix it to 0.0.0.0.

https://github.com/grafana/agent/blob/2603bda1660254f57f22e42ac3b47ab2ef58eb6e/internal/component/otelcol/extension/jaeger_remote_sampling/jaeger_remote_sampling.go#L41-L50

**Pull request merged**